lib/collection.js

```diff
@@ -60,7 +58,15 @@
 			}
 
 			// stringify data to json
-			if (options.data && _.isObject(options.data)) {
+			if (
+				options.data &&
+				_.isObject(options.data) &&
+				_([
+					'post', 'patch', 'put'
+				]).contains((options.type || params.type).toLowerCase())
```

Comment:
I think it's a good practice to split such huge constructions to small peaces to prevent unreadable `if` conditions.
In this case I'd define a type variable `var type = (options.type || params.type).toLowerCase();` above and change this line to
```
_(['post', 'patch', 'put']).contains(type)
```